src/server/pkg/storage/chunk/client.go

```diff
@@ -0,0 +1,172 @@
+package chunk
+
+import (
+	"context"
+	fmt "fmt"
+	io "io"
+	"io/ioutil"
+	"path"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/pachyderm/pachyderm/src/client/pkg/errors"
+	"github.com/pachyderm/pachyderm/src/server/pkg/obj"
+	"github.com/pachyderm/pachyderm/src/server/pkg/storage/tracker"
+)
+
+// Client allows manipulation of individual chunks, by maintaining consistency between
+// a tracker and an obj.Client.
+type Client struct {
```

Comment:
I still think the creation of temporary objects and renewal should be refactored into a general purpose abstraction in the tracker package.